pkg/apis/hive/v1/azure/platform.go

```diff
@@ -18,6 +18,11 @@ type Platform struct {
 
 	// BaseDomainResourceGroupName specifies the resource group where the azure DNS zone for the base domain is found
 	BaseDomainResourceGroupName string `json:"baseDomainResourceGroupName,omitempty"`
+
+	// ResourceGroupName is used to indicate that the cluster was installed into an already-existing
+	// Azure resource group (instead of having the installer create the resource group based on the
+	// cluster name). This will be used to indicate to the uninstall process where it should clean up.
```

Comment:
Let's make this description more user-focused.
```suggestion
	// ResourceGroupName specifies the name of the Azure resource group containing the cluster.
	// This must match the name of the Azure resource group specified in the InstallConfig.
	// If the name of the Azure resource group is omitted from the InstallConfig, then it must be
	// omitted here as well.
```